### DISCUSSION TITLE
uBOL adding dashboard suppression

### DISCUSSION THREAD

**Discussion by dluong06** [uBlockOrigin/uBOL-home#61] [Ideas]
As others have started doing, we wanted to get ahead of MV3 implantation in our environment. However first install launch of the dashboard may cause some users to freak out. 

Similar to this discussion, can I request that dashboard suppression be added? 
https://github.com/uBlockOrigin/uBOL-issues/discussions/35

**Comment by gorhill** [comment]
https://github.com/gorhill/uBlock/commit/b9f3523c9539ecf1c603d304742e1d010ff22a7a

**Comment by dluong06** [comment]
Noticed this is committed to 1.0.23.8155 in the public release in Chrome Web Store. In testing it does not seem to work for the public release but it does work in the dev build 1.51.1b23.

Added picture for registry entry ref.
![image](https://github.com/uBlockOrigin/uBOL-home/assets/140841773/3131dc27-d951-4d11-9452-75e8bfe86501)

**Reply by gorhill** [reply]
> it does work in the dev build 1.51.1b23.

1.51.1b23 is uBlock Origin, it does not support that setting. I implemented the setting for uBlock Origin Lite (uBO Lite).

The [Chrome store version of uBO Lite](https://chrome.google.com/webstore/detail/ublock-origin-lite/ddkjiahejlhfcafbddmgiahcphecmpfh) supports the setting. I tried it locally before publishing and it worked on my side as expected -- using linux. For windows, I wouldn't know how this works. The value for `disableFirstRunPage` is boolean, this won't work if it's a string.

**Reply by gorhill** [reply]
This [screenshot](https://user-images.githubusercontent.com/6779535/69871934-fc4f8980-12ab-11ea-8675-d0f0d26c25c4.png) found at [1] suggests you are missing the `policy` part under the extension id.

[1] `https://github.com/andy-portmen/open-in/issues/36#issuecomment-559789771`

---

The [documentation for windows](https://www.chromium.org/administrators/configuring-policy-for-extensions/#windows) also show usage of `policy`:

> `.../3rdparty\extensions\gihmafigllmhbppdfjnfecimiohcljba\policy`

**Reply by dluong06** [reply]
Ended up doing a few tests with the information you provided and got it to work. Here is the correct configuration for anyone else that may look this up in the future.

It would be a REG_DWORD with the value of 1 for True within the subkey of "policy" under "ddkjiahejlhfcafbddmgiahcphecmpfh" FOR WINDOWS MACHINES
Full path "Computer\HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Google\Chrome\3rdparty\extensions\ddkjiahejlhfcafbddmgiahcphecmpfh\Policy"

![image](https://github.com/uBlockOrigin/uBOL-home/assets/140841773/b9f5e251-6ef2-49c4-aaef-dc363e56a611)

**Reply by gorhill** [reply]
Looks like what worked for you is [explained here as _Option 3_](https://www.chromium.org/administrators/complex-policies-on-windows/#option-3-expanded-json-in-the-registry):

> Booleans are also stored as Dword Values. Use 0 for false and 1 for true.

---

So I suspect if you had used `"true"` instead of `true`, it would have worked (_Option 2_), since `JSON.stringify(true)` is `"true"`.

**Comment by dluong06** [comment]
Sorry to bring up this old thread again. MS Edge seems to not process the policy of disableFirstRunPage in the same manner as Chrome despite both being Chromium based. Not sure if this is related to how uBOL finds the policy or an Edge issue. 

I've tried this with both the Edge store uBOL and Chrome Store uBOL. I've also tested with diableFirstRunPage = 1 for true in both policy and the extension key itself. Attached screenshots of where MS Edge64 handles forced extension installs and basic testing of the policy. 

![image](https://github.com/uBlockOrigin/uBOL-home/assets/140841773/0861f64c-0f61-4a42-9524-78125db78c77)

![image](https://github.com/uBlockOrigin/uBOL-home/assets/140841773/47b3149f-c06a-4423-9aaf-7d67c97825a0)

**Reply by gorhill** [reply]
> how uBOL finds the policy

It's just a standard API, through [`browser.storage.managed.get()`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/storage/managed). I don't see documentation specific to Microsoft Edge for that API, they defer to the official Chrome documentation.

**Reply by lms-ay** [reply]
The extension ID for Edge is different from the one in Chrome.
HKLM\Software\Policies\Google\Chrome\3rdparty\extensions\ddkjiahejlhfcafbddmgiahcphecmpfh\policy
HKLM\Software\Policies\Microsoft\Edge\3rdparty\extensions\cimighlppcgcoapaliogpjjdehbnofhn\policy
